streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorNode.java

```diff
@@ -60,15 +74,111 @@ public void addChild(ProcessorNode<?, ?> child) {
         children.add(child);
     }
 
-    public void init(ProcessorContext context) {
+    public void init(ProcessorContext context, StreamsMetrics metrics) {
```

Comment:
Would it be better for `metrics` to be passed in the `ProcessorNode` constructor instead? For example, like it's done in the `StreamThread` constructor.